components/client/src/main/java/com/hotels/styx/client/OriginsInventory.java

```diff
@@ -88,7 +88,7 @@
     private static final HealthyEvent HEALTHY = new HealthyEvent();
     private static final UnhealthyEvent UNHEALTHY = new UnhealthyEvent();
 
-    private final Announcer<OriginsChangeListener> inventoryListeners = Announcer.to(OriginsChangeListener.class);
+    private final List<OriginsChangeListener> inventoryListeners = new CopyOnWriteArrayList<>();
```

Comment:
Please just move the `Announcer` class away from `styx-api` into `styx-common`. Then these other classes can still keep using it.

Reply:
Also, please note that the `Announcer` is used inside the API class `AbstractRegistry` so it cannot be moved without also moving that one.